drivers/arm/tzc/tzc_dmc500.c

```diff
@@ -271,5 +272,13 @@ void tzc_dmc500_driver_init(const tzc_dmc500_driver_data_t *plat_driver_data)
 	 */
 	/* Validates the information passed by platform */
 	validate_plat_driver_data(plat_driver_data);
-	g_driver_data = plat_driver_data;
+
+	g_driver_data = (tzc_dmc500_driver_data_t *)plat_driver_data;
+
+	/* Check valid system interface count */
+	assert(g_driver_data->sys_if_count <= MAX_SYS_IF_COUNT);
+
+	/* If interface count is not present then assume max */
+	if (g_driver_data->sys_if_count == 0U)
+		g_driver_data->sys_if_count = MAX_SYS_IF_COUNT;
```

Comment:
@amitdanielkachhap , I just realized this patch can sublty cause issues if the plat_driver_data is pointing to a const to begin with. The tzc_dmc500_driver_init() prototype is expecting a pointer to a const as the arg, and hence the compiler wouldn't flag error.
The solution preferred is to keep g_driver_data as const tzc_dmc500_driver_data_t *. Define another global variable, say g_sys_if_count, and initialize it to plat_driver_data->sys_if_count. Use that for the checks.

Reply:
@soby-mathew: Agreed. Sending a new pull request with this change.